packages/yarnpkg-fslib/sources/ZipOpenFS.ts

```diff
@@ -750,26 +772,30 @@ export class ZipOpenFS extends BasePortableFakeFS {
     });
 
     if (this.zipInstances) {
-      let zipFs = this.zipInstances.get(p);
+      let cachedZipFs = this.zipInstances.get(p);
 
-      if (!zipFs) {
+      if (!cachedZipFs) {
         const zipOptions = await getZipOptions();
 
         // We need to recheck because concurrent getZipPromise calls may
         // have instantiated the zip archive while we were waiting
-        zipFs = this.zipInstances.get(p);
-        if (!zipFs) {
-          zipFs = new ZipFS(p, zipOptions);
+        cachedZipFs = this.zipInstances.get(p);
+        if (!cachedZipFs) {
+          cachedZipFs = {
+            zipFs: new ZipFS(p, zipOptions),
+            time: Date.now(),
```

Comment:
Better to call `Date.now()` before the two conditions, and use the cached result both here and line 797